File: 1701/solution.py
class Solution:
    def averageWaitingTime(self, customers: list[list[int]]) -> float:
        total = 0
        time = 0
        for start, wait in customers:
            if time < start:
                time = start
            time += wait
            total += time - start
        return total / len(customers)

File: 1701/solution.java
class Solution {
    public double averageWaitingTime(int[][] customers) {
        double total = 0;
        int currTime = 0;
        for (int[] cust : customers) {
            if (currTime < cust[0]) {
                currTime = cust[0];
            }
            currTime += cust[1];
            total += currTime - cust[0];
        }
        return total / customers.length;
    }
}

// time O(n)
// space O(1)